examples/mlflow_artifacts/README.md

```diff
@@ -0,0 +1,45 @@
+This directory contains a set of files for demonstrating the MLflow Artifacts service.
+
+## Build a wheel
+
+This example requires a wheel for `mlflow` to be stored in `dist`.
+
+```sh
+# Clean up existing wheels
+rm dist/*
+
+# Build a wheel for the development version of mlflow
+pip wheel --no-deps --wheel-dir dist ../..
+
+# Build a wheel for the latest version of mlflow on PyPI
+pip wheel --no-deps --wheel-dir dist mlflow
+```
```

Comment:
Why are these steps required?

Reply:
This is mainly for development purposes. We can simplify the workflow by preparing two Docker files (`Dockerfile` and `Dockerfile.dev`). 

`docker-compose.yml` would look like:

```
# docker-compose.yml
  tracking-server:
    build:
      context: .
      # Defaults to `Dockerfile` if `DOCKERFILE` environment variable doesn't exist.
      dockerfile: "${DOCKERFILE:-Dockerfile}"
    depends_on:
      - postgres
      - artifacts-server
```

`README.md` would look like:

````md
This directory contains a set of files for demonstrating the MLflow Artifacts service.

## Run the example

```sh
# Build services
docker-compose build

# Launch tracking and artifacts servers
docker-compose up -d

# Run `run.py` that uploads, downloads, and list artifacts
docker-compose run -v ${PWD}/run.py:/app/run.py client python run.py
```

## Explore the logging results:

```sh
# Make sure both tracking and artifacts servers are running
docker-compose ps
```

- MLflow UI is available at http://localhost:5000 to explore tracking results.
- MinIO Console is available at http://localhost:9001 to explore logged artifacts. The login username and password are `user` and `password`.

## Reset tracking and artifacts servers

```
docker-compose down --volumes --remove-orphans
```

👇👇👇

## Development

```
pip wheel --no-deps --wheel-dir dist ../..
DOCKERFILE=Dockerfile.dev docker-compose build
docker-compose run -v ${PWD}/run.py:/app/run.py client python run.py
```

````

What do you think about this approach?

Reply:
Dockerfile:

```
FROM python:3.6

WORKDIR /app

RUN pip install psycopg2 boto3
RUN pip install mlflow
```

Dockerfile.dev:

```
FROM python:3.6

WORKDIR /app

RUN pip install psycopg2 boto3
RUN pip install mlflow
COPY dist ./dist
RUN pip install dist/mlflow-*.whl
```